src/test/java/com/xwiki/authentication/saml/XWikiGroupManagerTest.java

```diff
@@ -0,0 +1,164 @@
+package com.xwiki.authentication.saml;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.Logger;
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+import com.xpn.xwiki.objects.classes.BaseClass;
+import com.xpn.xwiki.web.XWikiRequest;
+import com.xpn.xwiki.web.XWikiResponse;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.slf4j.LoggerFactory;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.DocumentReferenceResolver;
+
+import javax.servlet.http.HttpSession;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class XWikiGroupManagerTest {
+    @Test
+    public void addUserToGroup_ShouldAddUserToGivenGroup() {
```

Comment:
```suggestion
    public void whenAddUserToGroup_ShouldAddUserToGivenGroup() {
```

Same for the other tests here.